### PR TITLE
Stable merge window for week 03 of 2020

### PR DESCRIPTION
**Pull request by matteodelabre**
New packages:

* decay 2.1\~beta-1

Updated packages:

* draft 0.2.0-16 → 0.2.0-17
    - Fix install script asking to disable shutdown.target
* erode 2.0.3\~beta-1 → 2.1\~beta-1
    - Add .oxide file
    - Automatically reload oxide on install
* fret 2.0.3\~beta-1 → 2.1\~beta-1
    - Add .oxide file
    - Automatically reload oxide on install
* oxide 2.0.3\~beta-1 → 2.1\~beta-1
    - Add .oxide file
    - Automatically reload oxide on install
* rot 1.0.0 → 2.1\~beta-1
    - Fix wrong version
* tarnish 2.0.3\~beta-1 → 2.1\~beta-1
    - Fix install script asking to disable shutdown.target
* koreader 2020.12-2 → 2020.12-3
    - Add .oxide file
    - Automatically reload oxide on install
    - Remove warning relative to Oxide
* plato 0.9.1-13 → 0.9.13-1
* genie 0.1.0-1 → 0.1.1-1
* simple 0.1.0-3 → 0.1.1-1
* remux 0.1.5-1 → 0.1.6-2
* toltec-bootstrap 0.0.1-2 → 0.0.1-3
    - Update wget to 1.21.1
* xochitl 0.0.0-2 → 0.0.0-3
    - Add .oxide file
    - Automatically reload oxide on install

Install:

* Update wget to 1.21.1
* Upgrade old opt.mount bind mount from remarkable_entware
* Keep branch from existing install if applicable

Documentation:

* Link to code of conduct and security policy.

Tooling:

* Add flag field to disable auto stripping of binaries
* Send Discord notifications on update

### CODE REVIEW

**Comment by matteodelabre** [comment]
Added the remux hotfix from #231.

**Comment by Eeems** [comment]
Has anybody had a chance to go through this?

**Comment by matteodelabre** [comment]
Not yet. I should have a free time window tomorrow afternoon to test this on rM1.

**Comment by matteodelabre** [comment]
Tested on rM1. No issues to report wrt the bootstrap script. Some issues with Oxide:

* KOReader still glitches Oxide, even after purging `~/.config/Eeems` (should this be related in any way).
* Killing plato from Oxide does not work (it stays in the background and interferes with Oxide).

**Comment by Eeems** [comment]
> 
>     * KOReader still glitches Oxide, even after purging `~/.config/Eeems` (should this be related in any way).

This shouldn't' be necessary if the oxide file is included. Can you confirm that the tarnish.conf contains the onpause and onresume lines for KOReader after installing?

Also, can you confirm that rot was upgraded properly? Or did it still assume v1.0.0 and thus not upgrade? If that's the case, it's never properly reloading the applications.

> 
>     * Killing plato from Oxide does not work (it stays in the background and interferes with Oxide).

Killing or just pausing it? I've had no issue with plato, so could this be an issue with the plato update?

File: package/draft/package [hunk 1]
```diff
@@ -5,7 +5,7 @@
 pkgnames=(draft)
 pkgdesc="Launcher which wraps around the standard interface"
 url=https://github.com/dixonary/draft-reMarkable
-pkgver=0.2.0-16
+pkgver=0.2.0-17
```
Comment (Eeems) [inline package/draft/package:8]: Why was this bumped?

**Comment by matteodelabre** [comment]
> This shouldn't' be necessary if the oxide file is included. Can you confirm that the tarnish.conf contains the onpause and onresume lines for KOReader after installing?

It does not. Here is the content of the `.config/Eeems/tarnish.conf` file:

```ini
[General]
version=1

[applications]
1\bin=/opt/bin/koreader
1\description="An ebook reader application supporting PDF, DjVu, EPUB, FB2 and many more formats"
1\displayName=KOReader
1\flags=system
1\icon=/opt/etc/draft/icons/koreader.png
1\name=KOReader
1\type=0
2\bin=/opt/bin/decay
2\description=Handles the password screen on resume
2\displayName=Startup application
2\flags=system, hidden
2\name=codes.eeems.decay
2\permissions=apps, power, system, wifi
2\type=0
3\bin=/opt/bin/erode
3\description=List and kill running processes
3\displayName=Process Manager
3\flags=system
3\name=codes.eeems.erode
3\type=0
4\bin=/opt/bin/fret
4\description=Handles taking screenshots
4\displayName=Screenshot daemon
4\flags=system, hidden, autoStart
4\name=codes.eeems.fret
4\permissions=notification, screen, system
4\type=1
5\bin=/opt/bin/oxide
5\description=Application launcher
5\displayName=Launcher
5\flags=system, hidden
5\name=codes.eeems.oxide
5\permissions=apps, wifi, power, system, notification
5\type=0
6\bin=/usr/bin/xochitl
6\description=Read documents and take notes
6\displayName=Xochitl
6\flags=system
6\icon=/opt/etc/draft/icons/xochitl.png
6\name=xochitl
6\type=0
size=6
```

Here are the steps I took to reproduce this:

* Starting from a clean install of stable (with empty .config/Eeems and .cache/Eeems dirs)
* Install Oxide 2.0.3~beta-1 and KOReader 2020.12-2
* Upgrade to Oxide 2.1~beta-1 and KOReader 2020.12-3 from this merge PR
* Run Oxide, start KOReader, pause KOReader with left button

> Also, can you confirm that rot was upgraded properly? Or did it still assume v1.0.0 and thus not upgrade? If that's the case, it's never properly reloading the applications.

rot was upgraded to package version `2.1~beta-1`. However `rot --version` still prints `rot 1.0.0`.

> Killing or just pausing it? I've had no issue with plato, so could this be an issue with the plato update?

Pausing it works fine. I can even launch another app like Xochitl and see no glitch. After I come back to Oxide and kill Plato from there, the two UIs start interfering with one another, and I can see a `plato` process lingering in `top`.

If I revert Plato to the old 0.9.1 version that’s currently in stable, I observe exactly the same effects, so it looks more like an Oxide bug.

**Comment by Eeems** [comment]
When you have it installed, does the oxide file for KOReader exist? And is it populated properly?

As for rot --version. Ignore that as I've been meaning to remove it.

As for Plato, I'll try to replicate on my device. But I'm very surprised that this is happening. Especially if pausing other applications works.

**Comment by matteodelabre** [comment]
> When you have it installed, does the oxide file for KOReader exist? And is it populated properly?

There is both a `.oxide` and a `.draft` file:

```
root@remarkable:~# cat /opt/usr/share/applications/KOReader.oxide
{
  "displayName": "KOReader",
  "description": "An ebook reader application supporting PDF, DjVu, EPUB, FB2 and many more formats",
  "bin": "/opt/bin/koreader",
  "icon": "/opt/etc/draft/icons/koreader.png",
  "type": "foreground",
  "events": {
    "pause": "/opt/koreader/fbdepth -d 16 -r 1",
    "stop": "/opt/koreader/fbdepth -d 16 -r 1",
    "resume": "/opt/koreader/fbdepth -d 8 -r 1"
  }
}
root@remarkable:~# cat /opt/etc/draft/koreader.draft
# Copyright (c) 2020 The Toltec Contributors
# SPDX-License-Identifier: MIT

name=KOReader
desc=An ebook reader application supporting PDF, DjVu, EPUB, FB2 and many more formats
imgFile=koreader
call=/opt/bin/koreader
term=killall -9 luajit
```

**Comment by Eeems** [comment]
From the top left menu could you select reload and then see if it properly populates?

**Comment by matteodelabre** [comment]
No it doesn’t. Could the typo in the following line (`evnets` instead of `events`) be the cause of this issue?

https://github.com/Eeems/oxide/blob/741d1c3029770335df4ef8bf9437dfc289923d7e/applications/system-service/appsapi.h#L576

**Comment by Eeems** [comment]
Okay, I just tried installing from this PR and it's registering koreader without the onpause/onresume/onstop stuff which is very odd since it worked when merging into testing. What's more odd is that other applications have empty entries for onpause/onresume/onstop.

Now, when I stop tarnish, remove tarnish.conf, and add it again it does register KOReader automatically. But without the onpause/onresume/onstop stuff.



> No it doesn’t. Could the typo in the following line (`evnets` instead of `events`) be the cause of this issue?
> 
> https://github.com/Eeems/oxide/blob/741d1c3029770335df4ef8bf9437dfc289923d7e/applications/system-service/appsapi.h#L576

That would do it. I'm surprised it was missed during the testing merge. I guess for a quick workaround for the stable merge we duplicate the events key with the misspelling.

**Comment by Eeems** [comment]
Okay, found the problem, it looks like there are various issues with ingesting oxide files around events that need to be resolved in a new release. I'm okay with merging as is, although maybe we should add the warning messages back to the KOReader package for now.

**Comment by matteodelabre** [comment]
After applying the hotfix, pausing KOReader works very well. It however suffers from the same issue as Plato when it is killed from Oxide.

Also, when resuming KOReader, the recalled screen does not have the right format. After the first redraw from KOReader, everything is back to normal. I assume this is due to the fact that `saveScreen()` is called after the `onPause` hook is invoked, and therefore the screen state is saved as if it was 16-bit, even though it actually is 8-bit?

2021-01-21T164126 424

**Comment by Eeems** [comment]
> Also, when resuming KOReader, the recalled screen does not have the right format. After the first redraw from KOReader, everything is back to normal. I assume this is due to the fact that `saveScreen()` is called after the `onPause` hook is invoked, and therefore the screen state is saved as if it was 16-bit, even though it actually is 8-bit?

Interesting, I'm surprised nobody else has reported this yet. Since there is a workaround (swipe from top, and then exit the menu to redraw) I'm fine with leaving this as-is for now 

> After applying the hotfix, pausing KOReader works very well. It however suffers from the same issue as Plato when it is killed from Oxide.

Still haven't been able to replicate on my device. I'm pausing the entire process group so it should work. https://github.com/Eeems/oxide/blob/master/applications/system-service/application.cpp#L81
Could you take a look at htop to see if something funky is happening there?

**Comment by Eeems** [comment]
Just as I say that I've replicated it with KOReader.

**Comment by Eeems** [comment]
Okay, when killing plato from oxide it does seem to draw to the screen. I think we leave this as an outstanding bug with oxide as it doesn't properly clear touch input on stop: https://github.com/Eeems/oxide/blob/master/applications/system-service/application.cpp#L186-L189

**Comment by Eeems** [comment]
Opened https://github.com/Eeems/oxide/issues/155

**Comment by Eeems** [comment]
@matteodelabre Ready to add hotfix.

**Review by Eeems — APPROVED** [review]
From my perspective this is ready to merge.

**Comment by matteodelabre** [comment]
I have not noticed any further issues on rM1. Testing on rM2 is still needed though.

**Comment by LinusCDE** [comment]
Tested. Testing method:

- Purging complete toltec install (.entware dir, opt/usr-share-zone mounts and all other packages that did changes).
- Rebooted to ensure everything runs without entware
- Installing toltec stable using the bootstrap script from this merge branch
- Explicitly installing/upgrading all the packages mentioned in the PR (built and transferred the ipkgs)

**Tarnish doesn't draw to screen anymore (remux, draft and xochitl start fine)**

Doing a `rm ~/.config/Eeems/*` had no effect. 


The logs seem all fine. Everything starts up without any errors. 

<details>
<summary>Logs (I don't see any errors but maybe it'll help)</summary>

```shell
reMarkable: ~/ sstop tarnish
reMarkable: ~/ sstart tarnish && journalctl -fu tarnish
-- Logs begin at Sun 2021-01-24 15:20:45 Europe. --
Jan 24 15:48:53 reMarkable tarnish[1560]: Removing all inhibitors
Jan 24 15:48:53 reMarkable tarnish[1560]: Wifi API false
Jan 24 15:48:53 reMarkable tarnish[1560]: Unregistering all networks
Jan 24 15:48:53 reMarkable tarnish[1560]: Unregistering all BSSs
Jan 24 15:48:53 reMarkable tarnish[1560]: Killing timer
Jan 24 15:48:53 reMarkable tarnish[1560]: Epaper integration dying
Jan 24 15:48:53 reMarkable tarnish[1560]: Epaper integration dead
Jan 24 15:48:53 reMarkable tarnish[1560]: Unlocked  /dev/input/event0
Jan 24 15:48:54 reMarkable systemd[1]: Stopped oxide system service.
Jan 24 15:49:01 reMarkable systemd[1]: Started oxide system service.
Jan 24 15:49:01 reMarkable tarnish[1797]: RM2 detected...
Jan 24 15:49:01 reMarkable tarnish[1797]: EPD platform plugin loaded!
Jan 24 15:49:01 reMarkable tarnish[1797]: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
Jan 24 15:49:01 reMarkable tarnish[1797]: Creating DBusService instance
Jan 24 15:49:01 reMarkable tarnish[1797]: Looking for wireless devices...
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking lo...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Not a wireless device
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking sit0...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Not a wireless device
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking usb0...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Not a wireless device
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking wlan0...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Wireless device found!
Jan 24 15:49:01 reMarkable tarnish[1797]: Loading networks from settings...
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering networks...
Jan 24 15:49:01 reMarkable tarnish[1797]:   Found network "XXXXXXXXX"
Jan 24 15:49:01 reMarkable tarnish[1797]: Loaded networks.
Jan 24 15:49:01 reMarkable tarnish[1797]: Turning wifi on
Jan 24 15:49:01 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:01 reMarkable tarnish[1797]: Reconnecting to wifi
Jan 24 15:49:01 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:01 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:01 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:01 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:01 reMarkable tarnish[1797]: Wifi API false
Jan 24 15:49:01 reMarkable tarnish[1797]: Auto Sleep 1
Jan 24 15:49:01 reMarkable tarnish[1797]: System API ready to use
Jan 24 15:49:01 reMarkable tarnish[1797]: Looking for batteries and chargers...
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking max77818-charger...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Found Charger!
Jan 24 15:49:01 reMarkable tarnish[1797]:   Checking max77818_battery...
Jan 24 15:49:01 reMarkable tarnish[1797]:     Found Battery!
Jan 24 15:49:01 reMarkable tarnish[1797]: "KOReader"
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/apps/67a9b43a3c265ebfb21ce90997e7b7d3" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: "codes.eeems.erode"
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/apps/f8c23b4485cf5760b3468cae491a0a72" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: "codes.eeems.fret"
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/apps/f3999e4363cb568ea6ccc50aa4b27b23" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: "xochitl"
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/apps/d941cc3512975cd9beb7dde71108afce" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: QDBusObjectPath: invalid path ""
Jan 24 15:49:01 reMarkable tarnish[1797]: QDBusObjectPath: invalid path ""
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering service...
Jan 24 15:49:01 reMarkable tarnish[1797]: Reading From :  /dev/input/event1  ( Wacom I2C Digitizer )
Jan 24 15:49:01 reMarkable tarnish[1797]: Listening for events...
Jan 24 15:49:01 reMarkable tarnish[1797]: Reading From :  /dev/input/event0  ( 30370000.snvs:snvs-powerkey )
Jan 24 15:49:01 reMarkable tarnish[1797]: locking  /dev/input/event0
Jan 24 15:49:01 reMarkable tarnish[1797]: /dev/input/event0  locked
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering exit handler...
Jan 24 15:49:01 reMarkable tarnish[1797]: Listening for keypresses...
Jan 24 15:49:01 reMarkable tarnish[1797]: Reading From :  /dev/input/event2  ( cyttsp5_mt )
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering object...
Jan 24 15:49:01 reMarkable tarnish[1797]: Connecting to deprecated signal QDBusConnectionInterface::serviceOwnerChanged(QString,QString,QString)
Jan 24 15:49:01 reMarkable tarnish[1797]: Listening for events...
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered
Jan 24 15:49:01 reMarkable tarnish[1797]: Unable to find current application
Jan 24 15:49:01 reMarkable tarnish[1797]: Launching  "/codes/eeems/oxide1/apps/f3999e4363cb568ea6ccc50aa4b27b23"
Jan 24 15:49:01 reMarkable tarnish[1797]: Application "codes.eeems.fret" is starting.
Jan 24 15:49:01 reMarkable tarnish[1797]: Application "codes.eeems.fret" is running.
Jan 24 15:49:01 reMarkable tarnish[1797]:  Waiting for tarnish to start up...
Jan 24 15:49:01 reMarkable tarnish[1797]: [codes.eeems.fret 1820] Requesting system API...
Jan 24 15:49:01 reMarkable tarnish[1797]: Checking permission "system" from requestAPI
Jan 24 15:49:01 reMarkable tarnish[1797]: "codes.eeems.fret" true
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering  "/codes/eeems/oxide1/system"
Jan 24 15:49:01 reMarkable tarnish[1797]: System API true
Jan 24 15:49:01 reMarkable tarnish[1797]: [codes.eeems.fret 1820] Requesting screen API...
Jan 24 15:49:01 reMarkable tarnish[1797]: Checking permission "screen" from requestAPI
Jan 24 15:49:01 reMarkable tarnish[1797]: "codes.eeems.fret" true
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering  "/codes/eeems/oxide1/screen"
Jan 24 15:49:01 reMarkable tarnish[1797]: Screen API true
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/screenshots/20210117T185657315" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/screenshots/20210117T185821079" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: Registered "/codes/eeems/oxide1/screenshots/20210124T154339414" codes.eeems.oxide1.Application
Jan 24 15:49:01 reMarkable tarnish[1797]: [codes.eeems.fret 1820] Requesting notification API...
Jan 24 15:49:01 reMarkable tarnish[1797]: Checking permission "notification" from requestAPI
Jan 24 15:49:01 reMarkable tarnish[1797]: "codes.eeems.fret" true
Jan 24 15:49:01 reMarkable tarnish[1797]: Registering  "/codes/eeems/oxide1/notification"
Jan 24 15:49:01 reMarkable tarnish[1797]: Notification API true
Jan 24 15:49:01 reMarkable tarnish[1797]: [codes.eeems.fret 1820] Connecting signal listener...
Jan 24 15:49:01 reMarkable tarnish[1797]: [codes.eeems.fret 1820] Waiting for signals...
Jan 24 15:49:04 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:04 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:04 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:04 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:07 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Jan 24 15:49:07 reMarkable tarnish[1797]: ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
^C
```

</details>

Also tried setting `LD_PRELOAD` manually and then starting tarnish by hand.

Not sure, but tarnish "seems" not to react to blind button pressed where apps should be.

**Minor: (Genie config [from here](https://remarkablewiki.com/faq/how_to_close_an_app?s[]=oxide) seems out of date with the current rot binary)**

<details>

<summary>Log</summary>

Started genie with config from wiki.
Did a 3 finger long tap then 2 finger long tap and a 1 finger long tap. (`jq` was installed as well.)

```shell
reMarkable: ~/ genie /opt/etc/genie.conf
USING /opt/etc/genie.conf AS CONFIG FILE
ADDED TAP GESTURE:
  command: /opt/bin/rot apps call previousApplication
  gesture: tap
  fingers: 1
  zone: -1 -1 -1 -1
  duration: 3
ADDED TAP GESTURE:
  command: /opt/bin/rot screen call screenshot
  gesture: tap
  fingers: 2
  zone: -1 -1 -1 -1
  duration: 3
ADDED TAP GESTURE:
  command: /opt/bin/rot apps get applications | /opt/bin/jq '."codes.eeems.erode"' -cr | sed 's|/codes/eeems/oxide1/||' | xargs -I {} /opt/bin/rot --object Application:{} apps
  gesture: tap
  fingers: 3
  zone: -1 -1 -1 -1
  duration: 3
RUNNING COMMAND /opt/bin/rot apps get applications | /opt/bin/jq '."codes.eeems.erode"' -cr | sed 's|/codes/eeems/oxide1/||' | xargs -I {} /opt/bin/rot --object Application:{} apps
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1.0.0' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Usage: /opt/bin/rot [options] api action
Oxide settings tool

Options:
  -h, --help             Displays this help.
  -v, --version          Displays version information.
  -o, --object <object>  Object to act on, e.g.
                         Network:network/94d5caa2d4345ab7be5254dfb9678cd7
  --once                 Exit on the first signal when listening.

Arguments:
  api                    wifi
                         power
                         apps
                         system
                         screen
                         notification
  action                 get
                         set
                         listen
                         call
RUNNING COMMAND /opt/bin/rot screen call screenshot
"/codes/eeems/oxide1/screenshots/20210124T154339414"
RUNNING COMMAND /opt/bin/rot apps call previousApplication
false
```

</details>

**Didn't get to test decay or any other oxide specifics that require it to run since tarnish didn't start.**

**Comment by Eeems** [comment]
As I mentioned on Discord, tarnish only draws to the screen when displaying a notification. The rest of the time you are seeing oxide or other apps running.

By the looks of the logs it looks like oxide and decay aren't installed as they aren't being registered. You can see what applications are registered with `rot apps get applications | jq`

**Review by LinusCDE — APPROVED** [review]
After a discussion on Discord, I think this should be ready for the rM 2 as well.

My issue was that I forgot to specify the decay .ipk file and for some reason opkg seemed to ignore it. On a proper repo the existing (checked it) dependency should prevent this from occurring in the first place.

Only known bug is that using "No pin" in decay has broken wakeup, but this I known and better than the current state of #241 .

**Pull request merged**